src/templates/DotnetTemplateRetriever.ts

```diff
@@ -48,13 +49,22 @@ export class DotnetTemplateRetriever extends TemplateRetriever {
         const itemFilePath: string = getDotnetItemTemplatePath(runtime);
         await downloadFile(cliFeedJson.releases[templateVersion].itemTemplates, itemFilePath);
 
-        this._rawTemplates = <object[]>JSON.parse(await executeDotnetTemplateCommand(runtime, undefined, 'list'));
-        return parseDotnetTemplates(this._rawTemplates, runtime);
+        return await this.parseTemplates(runtime);
+    }
+
+    protected async getTemplatesFromBackup(runtime: ProjectRuntime): Promise<IFunctionTemplate[]> {
+        await fse.copy(ext.context.asAbsolutePath(path.join('resources', 'backupDotnetTemplates')), getDotnetTemplatesPath(), { overwrite: true, recursive: false });
```

Comment:
create a getDotnetTemplatesBackupPath()?

Reply:
You're overwriting a location inside the installation location of the extension?  Or am I missing something?  That doesn't sound like a good idea.

Reply:
In terms of `getDotnetTemplatesBackupPath`, not a huge fan of creating simple functions that are only used once.

And yep - that's exactly what I'm doing. It's the same location we use when downloading these nuget packages directly from the cli-feed. If you're worried about overwriting, I'm doing it because the packages we downloaded from the cli-feed might be corrupt (otherwise why would we have gotten to the 'backup' case?)

If you're worried about using the installation location of the extension as a storage place in the first place, I'm doing it because we have to reference the nuget package every time the user creates a template. We can't just download the nuget package once and throw it away (otherwise I would've used the temp folder). `ExtensionContext` provides a workspace-specific storage path, but I'm pretty sure you're supposed to use the extension's install path for 'global' storage

Reply:
In general it's a no-no because users don't necessarily have write access to where apps are installed. But I believe these aren't stored in program files.

Reply:
> In terms of getDotnetTemplatesBackupPath, not a huge fan of creating simple functions that are only used once.

It's more about discoverability and organization, since the other locations are discoverable from where they're grouped in those simple functions.

Reply:
... but just consider it a suggestion

Reply:
Yeah these are the install locations:
* Windows `%USERPROFILE%\.vscode\extensions`
* macOS `~/.vscode/extensions`
* Linux `~/.vscode/extensions`

I'm open to storing this somewhere else if you have a better idea. Although I will say that another benefit of storing these in the extension's folder itself is that it will be cleaned up when un-installing the extension.